buttercup/cogs/history.py

```diff
@@ -858,7 +858,12 @@ async def _until(
         elif user:
             # Take the next rank for the user
             next_rank = get_next_rank(user["gamma"])
-            goal_gamma, goal_str = parse_goal_str(next_rank["name"])
+            if next_rank:
+                goal_gamma, goal_str = parse_goal_str(next_rank["name"])
+            else:
+                # If the user has reached the maximum rank, take the next 10,000 tier
+                goal_gamma = ((user["gamma"] + 10_000) // 10_000) * 10_000

```

Comment:
I thought "surely there's a better way to do this" and then I looked it up, and there is not a better way to do this. Works for me!

Reply:
It definitely felt stupid to write, but I couldn't think of a better way